src/assets/TokenBalancesController.ts

```diff
@@ -93,14 +90,14 @@ export class TokenBalancesController extends BaseController<TokenBalancesConfig,
     const assets = this.context.AssetsController as AssetsController;
     const { selectedAddress } = assets.config;
     const { tokens } = this.config;
-    const newContractBalances: { [address: string]: typeof BN } = {};
+    const newContractBalances: { [address: string]: BN } = {};
     for (const i in tokens) {
       const { address } = tokens[i];
       try {
         newContractBalances[address] = await assetsContract.getBalanceOf(address, selectedAddress);
         tokens[i].balanceError = null;
       } catch (error) {
-        newContractBalances[address] = 0;
+        newContractBalances[address] = new BN(0);
```

Comment:
Commenting for visibility: This is the sole functional change (excluding type changes).

I'll scan through mobile to try and get some idea whether this would have an impact before merging.

Reply:
Mobile appears to normalize this to a hex string using `.toString(16)` in most places. For a number the `16` is ignored, but it does successfully output "0".

I did find one example where this would fail though: https://github.com/MetaMask/metamask-mobile/blob/develop/app/util/accountSecurity.js#L38
The call to `.gt` would blow up if the balance was set to `0`.

These balances are used in many other places too - I stopped looking after this one.

cc @estebanmino: It looks like this might fix a few mobile bugs, but I don't anticipate this introducing any.